README.md

```diff
@@ -46,25 +47,88 @@ Based on the separation of the two stages, STORM is implemented in a highly modu
 
 
 
-## Getting started
+## Installation
 
-### 1. Setup
 
-Below, we provide a quick start guide to run STORM locally.
+To install the knowledge storm library, use `pip install knowledge-storm`. 
 
+You could also install the source code which allows you to modify the behavior of STORM engine directly.
 1. Clone the git repository.
-   ```shell
-   git clone https://github.com/stanford-oval/storm.git
-   cd storm
-   ```
+    ```shell
+    git clone https://github.com/stanford-oval/storm.git
+    cd storm
+    ```
    
 2. Install the required packages.
    ```shell
    conda create -n storm python=3.11
    conda activate storm
    pip install -r requirements.txt
    ```
-3. Set up OpenAI API key (if you want to use OpenAI models to power STORM) and [You.com search API](https://api.you.com/) key. Create a file `secrets.toml` under the root directory and add the following content:
+   
+
+## API
+The STORM knowledge curation engine is defined as a simple Python `STORMWikiRunner` class.
+
+As STORM is working in the information curation layer, you need to set up the information retrieval module and language model module to create a `STORMWikiRunner` instance. Here is an example of using You.com search engine and OpenAI models.
+```python
+import os
+from knowledge_storm import STORMWikiRunnerArguments, STORMWikiRunner, STORMWikiLMConfigs
+from knowledge_storm.lm import OpenAIModel
+from knowledge_storm.rm import YouRM
+
+lm_configs = STORMWikiLMConfigs()
+openai_kwargs = {
+    'api_key': os.getenv("OPENAI_API_KEY"),
+    'temperature': 1.0,
+    'top_p': 0.9,
+}
+# STORM is a LM system so different components can be powered by different models to reach a good balance between cost and quality.
+# For a good practice, choose a cheaper/faster model for `conv_simulator_lm` which is used to split queries, synthesize answers in the conversation. Choose a more powerful model for `article_gen_lm` to generate verifiable text with citations.
+gpt_35 = OpenAIModel(model='gpt-3.5-turbo', max_tokens=500, **openai_kwargs)
```

Comment:
should we keep this as different instance for each of the lm in lm_config? As it can help distinguish the token usage and call history by different module in the pipeline.

Reply:
Based on [how `log_execution_time_and_lm_rm_usage` is implemented](https://github.com/stanford-oval/storm/blob/dev-python-pkg/knowledge_storm/interface.py#L370), I think we already distinguish the token usage in different stage.

The current `collect_and_reset_lm_history` does not distinguish different modules though.